### PR TITLE
prov/verbs: Fix Klocwork critical issues

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

File: prov/verbs/src/ep_rdm/verbs_rdm_cm.c [hunk 1]
```diff
@@ -78,9 +78,8 @@ fi_ibv_rdm_batch_repost_receives(struct fi_ibv_rdm_conn *conn,
 {
 	const size_t idx = (conn->cm_role == FI_VERBS_CM_SELF) ? 1 : 0;
 	struct ibv_recv_wr *bad_wr = NULL;
-	struct ibv_recv_wr wr[num_to_post];
+	struct ibv_recv_wr wr[num_to_post + 1];
```
Comment (shefty) [inline prov/verbs/src/ep_rdm/verbs_rdm_cm.c]: Why would we want the array size larger than the number to post?

Comment (gladkovdmitry17) [reply prov/verbs/src/ep_rdm/verbs_rdm_cm.c]: My goal was to not add overhead and not [go beyond the WR's array](https://github.com/ofiwg/libfabric/blob/d61f9cb9043c85cc8c072d414f0ec8f0017d9328/prov/verbs/src/ep_rdm/verbs_rdm_cm.c#L103).
But I agree, it  doesn't look natural. I'll re-implement to make it more understandable

**Comment by shefty** [comment]
@a-ilango - can you look over these changes?

File: prov/verbs/src/ep_rdm/verbs_queuing.h [hunk 2]
```diff
@@ -188,6 +188,10 @@ fi_ibv_rdm_move_to_postponed_queue(struct fi_ibv_rdm_request *request)
 	if (dlist_empty(&conn->postponed_requests_head)) {
 		struct fi_ibv_rdm_postponed_entry *entry =
 			util_buf_alloc(fi_ibv_rdm_postponed_pool);
+		if (OFI_UNLIKELY(!entry)) {
+			VERBS_WARN(FI_LOG_EP_DATA, "Unable ");
```
Comment (a-ilango) [inline prov/verbs/src/ep_rdm/verbs_queuing.h]: The log string looks truncated.

File: prov/verbs/src/fi_verbs.c [hunk 3]
```diff
@@ -376,29 +376,35 @@ static int fi_ibv_get_param_int(char *param_name, char *param_str,
...
 
 	fi_param_define(&fi_ibv_prov, param_name, FI_PARAM_INT, param_help);
 
 	if (!fi_param_get_int(&fi_ibv_prov, param_name, &param))
 		*param_default = param;
 
+fn2:
 	free(param_help);
-	return 0;
+fn1:
```
Comment (a-ilango) [inline prov/verbs/src/fi_verbs.c]: What's fn1, fn2? err1, err2 are the error label conventions followed in rest of the code.

**Comment by shefty** [comment]
Moved commit to PR #3297 in order to make requested changes and move the merger forward.

**Pull request closed without merging**